squire_lib/src/pairings/greedy.rs

```diff
@@ -14,45 +14,47 @@ pub fn greedy_pairings(
         paired: Vec::with_capacity(plyrs.len() / match_size + 1),
         rejected: Vec::new(),
     };
+    plyrs.reverse();
```

Comment:
I see what you're doing here, but I think it would be wise to change the arguments to this function. There is no reason this method *needs* to take in players as a `Vec`. The only thing that we care about is having ownership over the IDs. So, we can change the function like so:
```rust
pub fn greedy_pairings<I>(mut plyrs: I, ... ) -> Pairings
  where I: Iterator<Item = PlayerId>
{ ... }
```
Why would we want to do this? It helps on two fronts. First, different pairing methods currently create an iterator, apply some transformation, collect into a vector, and then pass the IDs into here. This function would be more ergonomic if we those last two steps were encapsulated in this function. It would make it easier to choose a different data structure for manipulating player ID. Second, we this would allow for an easy switch to a `VecDeque`. This type of queue allows for pushes and pops in the front and back without a ton of extra copying. This will make the `reverse` call and the later `Iterator::rev` call moot